hisyonosuke/src/attendance-manager/attendanceManager.ts

```diff
@@ -356,6 +388,385 @@ const checkAttendance = (client: SlackClient, channelId: string) => {
+
+  // 時系列昇順に並び替え
+  return messages.reverse();
+}
+
+const getterForUserWorkStatusesByMessages = (messages: Message[], botUserId: string): (slackUserId: string) => UserWorkStatus | undefined => {
+  const processedMessages = getProcessedMessages(messages, botUserId);
+
+  // TODO: ↓ 「今誰いる？」の機能に流用する
+  const clockedInUserIds = Array.from(new Set(processedMessages.map(message => message.user)));
+  const clockedInUserWorkStatuses = clockedInUserIds.map(userSlackId => {
+    const userMessages = processedMessages.filter(message => message.user === userSlackId);
+    const userCommands = userMessages.map(message => getCommandType(message)).filter(_ => _);
+    const workStatus = getUserWorkStatusByLastCommand(userCommands[userCommands.length - 1]);
+    const needTrafficExpense = checkTrafficExpense(userCommands);
+
+    return {
+      userSlackId,
+      needTrafficExpense,
+      workStatus,
+      processedMessages: userMessages
+    }
+  });
+
+  return (userSlackId: string) => {
+    const matchedUserWorkStatus = clockedInUserWorkStatuses.filter(userWorkStatus => userWorkStatus.userSlackId === userSlackId);
+    if (matchedUserWorkStatus.length === 0) {
+      return undefined;
+    } else if (matchedUserWorkStatus.length === 1) {
+      return matchedUserWorkStatus[0];
+    } else {
+      throw new Error(); //TODO: エラーメッセージ
+    }
+  }
+}
+
+const getProcessedMessages = (messages: Message[], botUserId: string) => {
+  const messagesWithoutError = messages.filter(message => {
+    return !message.reactions?.filter(reaction => {
+      return (
+        reaction.users.includes(botUserId) &&
+        reaction.name === REACTION.ERROR
+      );
+    }).length
+  });
+  const processedMessages = messagesWithoutError.filter(message => {
+    return message.reactions?.filter(reaction => {
+      return (
+        reaction.users.includes(botUserId) &&
+        [
+          REACTION.DONE_FOR_TIME_RECORD,
+          REACTION.DONE_FOR_REMOTE_MEMO,
+          REACTION.DONE_FOR_LOCATION_SWITCH
+        ].includes(reaction.name)
+      )
+    }).length
+  });
+  return processedMessages;
+}
+
+const getUnprocessedMessages = (messages: Message[], botUserId: string) => {
+  const messagesWithoutError = messages.filter(message => {
+    return !message.reactions?.filter(reaction => {
+      return (
+        reaction.users.includes(botUserId) &&
+        reaction.name === REACTION.ERROR
+      );
+    }).length
+  });
+  const unprocessedMessages = messagesWithoutError.filter(message => {
+    return !message.reactions?.filter(reaction => {
+      return (
+        reaction.users.includes(botUserId) &&
+        [
+          REACTION.DONE_FOR_TIME_RECORD,
+          REACTION.DONE_FOR_REMOTE_MEMO,
+          REACTION.DONE_FOR_LOCATION_SWITCH
+        ].includes(reaction.name)
+      )
+    }).length
+  });
+  return unprocessedMessages;
+}
+
+const checkTrafficExpense = (userCommands: CommandType[]) => {
+  // 「リモート出勤」よりあとに「出社」がなければ交通費はかからなず、それ以外は必要
```

Comment:
```suggestion
  // 「リモート出勤」よりあとに「出社」がなければ交通費はかからず、それ以外は必要
```
誤字っぽい。「かからず」を意図している？

Reply:
誤字でした 😂